recipes/hdf5/meta.yaml

```diff
@@ -0,0 +1,36 @@
+package:
+  name: hdf5
+  version: 1.8.16
+
+source:
+  fn: hdf5-1.8.16.tar.gz
+  url: http://www.hdfgroup.org/ftp/HDF5/current/src/hdf5-1.8.16.tar.gz
+  md5: b8ed9a36ae142317f88b0c7ef4b9c618
+
+
+build:
+  features:
+    - vc9     # [win and py27]
+    - vc10    # [win and py34]
+    - vc14    # [win and py35]
+  detect_binary_files_with_prefix: True
+
+requirements:
+  build:
+    - python    # [win]
```

Comment:
It's a bit of black magic, but the Python build dependency combined with the `vc` features is just trying to enforce that we use the same VC version as the one CPython recommends to build that version of CPython with. For instance, VC9 is recommend with Python 2.7. So, we use these two together.

In practice, this seems to be too hard of a constraint especially when people want C++11 support and want to work with Python 2.7, for instance, which has a recommended compiler that is very old and lacks C++11 functionality. There are other examples.

However, I don't think it will be a problem for us to relax this constraint going forward. Namely, using a newer VC with CPython or potentially using some other VC version. There also appears to be significant demand for this functionality in the Conda Community.